tests/DataFactories/Settings.php

```diff
@@ -32,6 +33,7 @@ function withConfirmationEmailBody($body = null) {
 
   function withConfirmationEmailEnabled() {
     $this->settings->set('signup_confirmation.enabled', '1');
+    FormModel::updateSuccessMessages();
```

Comment:
I am not sure this side effect is a positive thing here. Data Factories should be really low level. And they shouldn't do any magical updates. Because if you want to test something and set up something you should be explicit in your test. First of all the test reads better when everything is explicit and secondly, you sometimes need to test different scenarios and this magic prevents that.

Reply:
Should I call `FormModel::updateSuccessMessages();` directly from within the tests or add a method `withUpdatedSuccessMessages` to the forms factory and call it instead?